src/heretic/model.py

```diff
@@ -38,14 +38,18 @@ def __init__(self, settings: Settings):
         print()
         print(f"Loading model [bold]{settings.model}[/]...")
 
-        self.tokenizer: PreTrainedTokenizerBase = AutoTokenizer.from_pretrained(
-            settings.model
-        )
+        try:
+            self.tokenizer: PreTrainedTokenizerBase = AutoTokenizer.from_pretrained(
+                settings.model
+            )
+        except Exception as error:
+            raise Exception(f"Failed to load tokenizer: {error}")
 
         # Fallback for tokenizers that don't declare a special pad token.
         if self.tokenizer.pad_token is None:
             self.tokenizer.pad_token = self.tokenizer.eos_token
-            self.tokenizer.padding_side = "left"
+            if hasattr(self.tokenizer, "padding_side"):
```

Comment:
There is still stuff unrelated to notebooks in this PR. All changes to `model.py` need to go.